src/utils/dump-context.js

```diff
@@ -32,20 +47,78 @@ function tstamp() {
   return retstr;
 }
 
-async function dump(context, action, index) {
-  const nowStr = tstamp();
-  action.debug = action.debug || {};
-  if (!action.debug.dumpDir) {
-    const id = (context.request && context.request.headers && context.request.headers['x-openwhisk-activation-id']) || '';
-    const pathStr = context.request && context.request.path ? context.request.path.replace(/\//g, '-').substring(1) : '';
-    const dirName = `context_dump_${nowStr}_${pathStr}_${id}`;
-    action.debug.dumpDir = path.resolve(process.cwd(), 'logs', 'debug', dirName);
-    await fs.ensureDir(action.debug.dumpDir);
+/**
+ * Ensures that the dump directory exists.
+ * @param {*} context Pipeline context
+ * @param {*} action Pipeline action
+ */
+async function ensureDumpDir(context, action) {
+  if (action.debug.dumpDir) {
+    return;
+  }
+  const stamp = tstamp(new Date());
+  const id = (context.request && context.request.headers && context.request.headers['x-openwhisk-activation-id']) || '';
+  const pathStr = context.request && context.request.path ? context.request.path.replace(/\//g, '-').substring(1) : '';
+  const dirName = `context_dump_${stamp}_${pathStr}_${id}`;
+  action.debug.dumpDir = path.resolve(process.cwd(), 'logs', 'debug', dirName);
+  await fs.ensureDir(action.debug.dumpDir);
+  action.logger.info(`Writing context dumps to: ${action.debug.dumpDir}`);
+}
+
+/**
+ * Writes the context dump to a file. If the dump directory doesn't exists, it will be created.
+ * @param {*} context Pipeline context
+ * @param {*} action Pipeline action
+ * @param {*} info The context dump information
+ */
+async function writeDump(context, action, info) {
+  const stamp = tstamp(info.time);
+  const fileName = `context-${stamp}-step-${info.index}-${info.name}.json`;
+  info.file = path.resolve(action.debug.dumpDir, fileName);
+  action.logger.silly(`writing context dump ${fileName}`);
+  return fs.writeFile(info.file, info.json, 'utf-8');
+}
+
+/**
+ * Records the current context dump in the {@code action.debug.contextDumps} array.
+ * If the log level is set to {@code silly}, the dump will also be written to disk using
+ * {@link #writeDump}.
+ *
+ * @param {*} context Pipeline context
+ * @param {*} action Pipeline action
+ * @param {number} index The index of the current pipeline step
+ * @param {string} name The name of the current pipeline step
+ */
+async function record(context, action, index, name) {
+  const debug = setdefault(action, 'debug', { contextDumps: [] });
+  const info = {
+    index,
+    name: name ? name.split(':').pop() : 'undef',
+    time: new Date(),
+    json: JSON.stringify(context, null, 2),
+  };
+  debug.contextDumps.push(info);
+
+  if (action.logger.level === 'silly' && !disableDisk()) {
```

Comment:
From the issue:

> always dump it for log-level debug and silly

Only `silly` seems to have been implemented.

Reply:
yeah, I though that maybe with `debug` we pollute too much. from my own experience, I often use `debug`, but then I guess, I don't need to dumps very often.....

but I'm indifferent.

Reply:
I am fine with `silly` only, just wanted to mention it...

Reply:
`silly` is fine.